pkg/acsengine/ssh.go

```diff
@@ -0,0 +1,51 @@
+package acsengine
+
+import (
+	"crypto/rand"
+	"crypto/rsa"
+	"fmt"
+
+	log "github.com/Sirupsen/logrus"
+	"golang.org/x/crypto/ssh"
+)
+
+const (
+	// SshKeySize is the size of SSH key to create
+	SSHKeySize = 4096
+)
+
+// CreateSaveSSH generates and stashes an SSH key pair.
+func CreateSaveSSH(username, outputDirectory string) (privateKey *rsa.PrivateKey, publicKeyString string, err error) {

```

Comment:
Possibly to get some tests around this too, even something as simple checking method signature + output?